client/extensions/wp-super-cache/state/stats/selectors.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Returns true if we are generating stats for the specified site ID, false otherwise.
+ *
+ * @param  {Object}  state Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {Boolean} Whether stats are being generated
+ */
+export function isGeneratingStats( state, siteId ) {
+	return get( state, [ 'extensions', 'wpSuperCache', 'stats', 'generateStatus', siteId, 'generating' ], false );
```

Comment:
Thanks for calling me out on this. I wasn't happy with how the selectors were implemented and your comment forced me to finally change it - https://github.com/Automattic/wp-calypso/pull/13705/commits/16164b08012c07f3ec80984ea4d6ed74715a35ec . :wink: